.github/workflows/ci-pr-docs.yml

```
@@ -0,0 +1,52 @@
+name: ci pr docs
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+on:
+  # TODO: change to pull_request when docusaurus site is ready
+  # pull_request:
+  push:
+    branches:
+      - idea404/docusaurus-docs
+    paths:
+      - 'docs/site/**'
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    outputs:
+      url: ${{ steps.deploy.outputs.deployment_url }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Install PNPM
+        uses: pnpm/action-setup@v3
+        with:
+          version: 8
+      - name: Install Vercel CLI
+        run: pnpm install --global vercel@latest
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Build Project Artifacts
+        run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Deploy Project Artifacts to Vercel
+        id: deploy
+        run: echo "::set-output name=deployment_url::$(vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }})"
```

`::set-output` is deprecated. Use the `$GITHUB_OUTPUT` variable instead. 
```suggestion
        run: echo "deployment_url=$(vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }})" >> $GITHUB_OUTPUT
```